File: types/analytics.ts
export interface FilterState {
  startDate: string;
  endDate: string;
  outlets: string[];
  brands: string[];
  categories: string[];
  orderTypes: string[];
  settlements: string[];
  searchTerm: string;
}

export interface SummaryMetrics {
  totalRows: number;
  totalRevenue: number;
  uniqueOrders: number;
  totalQuantity: number;
  aov: number;
  avgItemPrice: number;
  dateRange: {
    min: string;
    max: string;
  };
  outlets: string[];
  brands: string[];
  categories: string[];
  orderTypes: string[];
  settlements: string[];
  totalItems: number;
}

export interface DailyTrend {
  date: string;
  revenue: number;
  orders: number;
  quantity: number;
}

export interface MonthlyTrend {
  month: string;
  revenue: number;
  orders: number;
  quantity: number;
}

export interface WeeklyTrend {
  week: string;
  revenue: number;
  orders: number;
  quantity: number;
}

export interface OutletMetric {
  outlet: string;
  revenue: number;
  orders: number;
  quantity: number;
  percentage: number;
}

export interface CategoryMetric {
  category: string;
  revenue: number;
  orders: number;
  quantity: number;
  percentage: number;
}

export interface OrderTypeMetric {
  orderType: string;
  revenue: number;
  orders: number;
  quantity: number;
  percentage: number;
}

export interface SettlementMetric {
  settlement: string;
  revenue: number;
  orders: number;
  quantity: number;
  percentage: number;
}

export interface ItemMetric {
  item: string;
  category: string;
  revenue: number;
  quantity: number;
  orders: number;
  avgPrice: number;
}

export interface BusinessInsightsData {
  topOutlet: { name: string; revenue: number; percentage: number };
  topCategory: { name: string; revenue: number; percentage: number };
  bestSellingItem: { name: string; category: string; revenue: number; quantity: number };
  mostUsedOrderType: { name: string; orders: number; percentage: number };
  peakRevenuePeriod: { date: string; revenue: number; orders: number };
  aov: number;
}

export interface AnalyticsData {
  daily: DailyTrend[];
  monthly: MonthlyTrend[];
  weekly: WeeklyTrend[];
  byOutlet: OutletMetric[];
  byCategory: CategoryMetric[];
  byOrderType: OrderTypeMetric[];
  bySettlement: SettlementMetric[];
  byItem: ItemMetric[];
  insights: BusinessInsightsData;
}

export interface OrderRecord {
  billNo: number;
  orderDate: string;
  outlet: string;
  brand: string;
  category: string;
  item: string;
  orderType: string;
  price: number;
  quantity: number;
  revenue: number;
  settlement: string;
}

export interface AggregatedMatrixCell {
  d: string; // Date
  o: string; // Outlet
  b: string; // Brand
  g: string; // Category / Group
  t: string; // Order Type
  s: string; // Settlement
  r: number; // Revenue
  n: number; // Unique Orders count
  q: number; // Quantity
  c: number; // Line item count (Records)
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel, Field
from typing import List, Optional

class AnalyticsFilters(BaseModel):
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    outlet: Optional[str] = None
    brand: Optional[str] = None
    group: Optional[str] = None
    order_type: Optional[str] = None
    settlement: Optional[str] = None

class SummaryResponse(BaseModel):
    total_revenue: float
    total_orders: int
    total_items_sold: float
    average_order_value: float
    total_outlets: int
    total_products: int

class RevenueTrendPoint(BaseModel):
    date: str
    revenue: float
    orders: int

class OutletPerformance(BaseModel):
    outlet_name: str
    revenue: float
    orders: int
    items_sold: float

class GroupPerformance(BaseModel):
    group: str
    revenue: float
    orders: int
    items_sold: float
    percentage: float

class OrderTypeDistribution(BaseModel):
    order_type: str
    revenue: float
    orders: int

class TopProduct(BaseModel):
    item: str
    group: str
    revenue: float
    quantity: float

class SettlementDistribution(BaseModel):
    settlement: str
    revenue: float
    orders: int

class OrderRecord(BaseModel):
    id: int
    BillNo: str
    Outlet_Name: str
    Brand: str
    Order_Datetime: str
    Group: str
    Item: str
    Price: float
    Quantity: float
    Revenue: float
    Order_Type: str
    Settlement: str

class PaginatedOrdersResponse(BaseModel):
    total_records: int
    page: int
    page_size: int
    total_pages: int
    orders: List[OrderRecord]

class FilterOptionsResponse(BaseModel):
    outlets: List[str]
    brands: List[str]
    groups: List[str]
    order_types: List[str]
    settlements: List[str]
    min_date: str
    max_date: str
